### PR TITLE
#180: Confirmation dialog on back with unsaved changes & edit fix

### PR DESCRIPTION
**Pull request by afijal**

### CODE REVIEW

File: src/components/Editor/index.js [hunk 1]
```diff
@@ -75,7 +87,8 @@ export default class Editor extends Component {
                     solution,                     
                     editorPosition: response.data.steps.length -1,
                     theActiveMathField: field,
-                    readOnly: this.props.match.params.action == 'view'
+                    readOnly: this.props.match.params.action == 'view',
+                    stepsFromLastSave: JSON.parse(JSON.stringify(response.data.steps))
```
Comment (oskarhinc) [inline src/components/Editor/index.js]: is JSON.parse(JSON.stringify(...)) needed here?

File: src/components/Editor/index.js [hunk 2]
```diff
@@ -405,23 +420,66 @@ export default class Editor extends Component {
         googleAnalytics('Save');
         axios.put(`${config.serverUrl}/solution/`, this.state.solution)
             .then(response => {
-                this.setState({editLink: config.serverUrl + '/problem/edit/' + this.state.solution.editCode})
+                this.setState({
+                    editLink: config.serverUrl + '/problem/edit/' + this.state.solution.editCode,
+                    stepsFromLastSave: JSON.parse(JSON.stringify(this.state.solution.steps))
```
Comment (oskarhinc) [inline src/components/Editor/index.js]: same here

File: src/strings.js [hunk 3]
```diff
@@ -90,7 +88,10 @@ class Locales {
             hash: "#",
             equation: "Equation",
             annotation: "Annotation",
-            share_link: "Share Link: "
+            share_link: "Share Link: ",
+            confirmation_modal_unsaved_title: "You have unsaved changes, do you wish to save?",
+            discard_changes: " Discard Changes",
+            save_changes: " Save CHanges"
```
Comment (oskarhinc) [inline src/strings.js]: there is unwanted capital H

File: src/components/Editor/index.js (hunk 1 above)
Comment (afijal) [reply src/components/Editor/index.js]: yeah, to clone an array without any references

**Pull request merged**